ddl/ddl_api.go

```diff
@@ -3400,6 +3400,26 @@ func CheckModifyTypeCompatible(origin *types.FieldType, to *types.FieldType) (al
 			// Changing integer to other types, reorg is absolutely necessary.
 			return unsupportedMsg, errUnsupportedModifyColumn.GenWithStackByArgs(unsupportedMsg)
 		}
+	case mysql.TypeFloat, mysql.TypeDouble:
+		switch to.Tp {
+		case mysql.TypeFloat, mysql.TypeDouble:
+			skipSignCheck = true
+			skipLenCheck = true
+		case mysql.TypeDate, mysql.TypeDatetime, mysql.TypeTimestamp, mysql.TypeEnum, mysql.TypeSet:
```

Comment:
For the current implementation of convertToEnum/convertToSet, there's no particular conversion from float to enum/set.

The default logic will cast float to unsigned int, and require the int value should less than the number of enum elements(very confusion), which may encounter such two occasions:
1. negative float may get error
2. positive float will lose fraction part, and the value NEED to be less than the number of elements
    - (e.g. column a = 111.1, `alter table t modify a enum(111.1)` will get error because 111 > 1(only one enum element))

Hence user cannot get convincing result when they try to modify float column to enum. Maybe it's better to open the limitation after we give properly implements about the conversions.

Reply:
great!